src/lib.rs

```diff
@@ -0,0 +1,450 @@
+extern crate sxd_document;
+
+use std::str;
+use std::string::String;
+use std::collections::HashMap;
+
+use sxd_document::{
+    parser,
+    Package,
+};
+use sxd_document::dom::{
+    Document,
+    Root,
+    ChildOfRoot,
+    Element,
+    ChildOfElement,
+};
+
+#[derive(Debug)]
+pub struct Program {
+    pub groups: Vec<StatementBody>
+}
+
+#[derive(PartialEq, Debug)]
+pub struct StatementBody {
+    pub blocks: Vec<Block>
+}
+
+#[derive(PartialEq, Debug)]
+pub struct Block {
+    pub block_type: String,
+    pub id: String,
+    pub fields: Option<HashMap<String, FieldValue>>,
+    pub statements: Option<HashMap<String, StatementBody>>,
```

Comment:
I see two potential reasons why you might have used `Option`s here. 

> None represents an absence of data, so if we don't have any statements, then `statements` should be set to `None`

This is true. However, an empty `HashMap` would also represent the idea. Neither approach is necessarily superior from a logical perspective, but just initializing with an empty `HashMap` here will simplify the API a great deal.

> There may not be any statements, in which case I would never have to build a `HashMap` at all. This way I only have to create one if I need it.

The only time I can think of when you would want to use an `Option` to delay initialization (assuming initialization can even be avoided) is if the type being wrapped with `Option` is expensive to create. A `HashMap` is not, so by paying that initial cost of initializing it (a completely negligible cost) you can make your life way easier.

Reply:
Thanks for pointing that out. I was aiming for the first reason you mentioned, because a block from Blockly XML might not include any `<statement>` children. But application developers will know which of their blocks should have statements, so making the API more complex just makes it harder to get to those statements. I'll remove the `Option` wrapper. (Same for fields, by the way.)